### DISCUSSION TITLE
fetchJSON response.body should be consistent

### DISCUSSION THREAD

**Discussion by Soesah** [ing-bank/lion#2184] [Ideas]
Not every API is set up to return `"Content-Type":"application/json"`, some return plain/text. I did a test to see how own ajax wrapper package would react:
![76a6044b-f908-4a99-b83a-60f8cbc9d624](https://github.com/ing-bank/lion/assets/2736210/0c22ce9b-6091-4c49-80cf-10a10d01b8ca)

The `body` on the response reflects the API value correctly, as you can see, but this creates a mismatch from what I would expect from `fetchJson`. I expect JSON, not a string.

Is this something that lion should handle, and perhaps throw if `response.body` cannot be parsed as JSON, or should this be picked up in our own ajax wrapper package?

**Comment by thepassle** [comment]
Please dont leak internal package names :)
